### DISCUSSION TITLE
Suggestions to create a terminal app with a fixed status bar

### DISCUSSION THREAD

**Discussion by charlbury** [Textualize/rich#3435] [General]
Can anyone point me in the direction of using rich or textualize to control my running code output?

I run loads of python scripts that run endlessly to collect data or stats or process files. Each one is a python script that print()s their progress to the terminal/console to show what they are currently doing and summarise the amount of work done to date.

It would be much better to have one static line at the bottom of the terminal window that was a static status bar on which I could put summary info about the running task like Total Requests, Failed Request Count, Fail Percentage, Current Batch ID, Last Batch ID etc. etc. For example, something like the below:

```
-----------------------------------------------------------------------------
V:1.7 | PCID:XYZ123 | Curr Batch: XDC-123             Calls: 259 | Fails: 12%
```

The general output from the app would continue to be displayed above this as a new update is printed to the console and I'd want it to scroll to the show the latest line printed with the option to scroll back through the history like a usual python app that prints to the terminal.

Can rich do this perhaps using the console.status class?

**Comment by willmcgugan** [comment]
You can do that with rich.live.Live. If you want anything more advanced that what Live offers, you might want to look at Textual https://textual.textualize.io/

**Reply by charlbury** [reply]
Does that need a live on the console and on the status class?

**Reply by willmcgugan** [reply]
You won't need to use the Status class. Just Live.

**Reply by charlbury** [reply]
I'm sorry, I, can't figure it out... Sorry. I can only figure the following out:

```python
from rich.status import Status
from time import sleep

rows = 100
with Status("Starting...") as status:
    for row in range(1, rows):
        print(f"Step {row}.")
        status.update(f"Processing... Step: {row} of {rows} ({row/rows*100:.0f}%)")
        sleep(0.3)
```
